### PR TITLE
Apply `scalafmt` and simplify the code a bit.

### PR DESCRIPTION
**Pull request by snadorp**
I'll update this PR once #48 is merged, as `Medium` is also misformatted.

### CODE REVIEW

**Comment by snowplowcla** [comment]
Confirmed! @snadorp  has signed the Individual Contributor License Agreement. Thanks so much

**Comment by snadorp** [comment]
Build failure is expected.

File: .travis.yml [hunk 1]
```diff
@@ -5,7 +5,7 @@ scala:
 jdk:
 - oraclejdk8
 script:
-- sbt coverage test coverageReport
+- sbt coverage test coverageReport scalafmtCheck
```
Comment (BenFradet) [inline .travis.yml]: should we do it first? no need to run the tests if the formatting isn't right, wdyt?

File: src/main/scala/com/snowplowanalytics/refererparser/Referer.scala [hunk 2]
```diff
@@ -20,10 +20,10 @@ package com.snowplowanalytics.refererparser
...
-sealed abstract trait Referer
-final case object UnknownReferer                                        extends Referer
-final case class SearchReferer  (source: String, term: Option[String])  extends Referer
-final case object InternalReferer                                       extends Referer
-final case class SocialReferer  (source: String)                        extends Referer
-final case class EmailReferer   (source: String)                        extends Referer
-final case class PaidReferer    (source: String)                        extends Referer
+sealed trait Referer
+case object UnknownReferer                                     extends Referer
+case class SearchReferer(source: String, term: Option[String]) extends Referer
```
Comment (BenFradet) [inline src/main/scala/com/snowplowanalytics/refererparser/Referer.scala]: can we make the case classes final?

File: .travis.yml (hunk 1 above)
Comment (snadorp) [reply .travis.yml]: Sure, will move it to the front.

File: src/main/scala/com/snowplowanalytics/refererparser/Referer.scala (hunk 2 above)
Comment (snadorp) [reply src/main/scala/com/snowplowanalytics/refererparser/Referer.scala]: oops, my bad, removed too many finals :)

**Pull request merged**